htdocs/admin/mails.php

```diff
@@ -737,26 +737,28 @@ function initfields()
 
 		print '<tr class="oddeven"><td>'.$langs->trans('MAIN_MAIL_DEFAULT_FROMTYPE').'</td>';
 		print '<td>';
-		if ($conf->global->MAIN_MAIL_DEFAULT_FROMTYPE === 'robot') {
-			print $langs->trans('RobotEmail');
-		} elseif ($conf->global->MAIN_MAIL_DEFAULT_FROMTYPE === 'user') {
-			print $langs->trans('UserEmail');
-		} elseif ($conf->global->MAIN_MAIL_DEFAULT_FROMTYPE === 'company') {
-			print $langs->trans('CompanyEmail').' '.dol_escape_htmltag('<'.$mysoc->email.'>');
-		} else {
-			$id = preg_replace('/senderprofile_/', '', $conf->global->MAIN_MAIL_DEFAULT_FROMTYPE);
-			if ($id > 0) {
-				include_once DOL_DOCUMENT_ROOT.'/core/class/emailsenderprofile.class.php';
-				$emailsenderprofile = new EmailSenderProfile($db);
-				$emailsenderprofile->fetch($id);
-				print $emailsenderprofile->label.' '.dol_escape_htmltag('<'.$emailsenderprofile->email.'>');
+		if (isset($conf->global->MAIN_MAIL_DEFAULT_FROMTYPE)) {
```

Comment:
This test is not similar, It will never reach the else.

Reply:
How can I preg_replace if $conf->global->MAIN_MAIL_DEFAULT_FROMTYPE isn't set ?
I guess I have to initialise it, but how sould I do ?

Reply:
When a value is not set and PHP need to process it, PHP autoconvert it into 0 or ''
So here, if we want to change without knowing what the code do, we should replace the 
`if ($conf->global->MAIN_MAIL_DEFAULT_FROMTYPE === 'robot') {`
into
`if (isset($conf->global->MAIN_MAIL_DEFAULT_FROMTYPE) && $conf->global->MAIN_MAIL_DEFAULT_FROMTYPE === 'robot') {`

Reply:
Ok, got it for if condition. But then, we will reach the else. How preg_replace() will manage an unset value ?
Sould we do this :
preg_replace('/senderprofile_/', '', isset($conf->global->MAIN_MAIL_DEFAULT_FROMTYPE) ? $conf->global->MAIN_MAIL_DEFAULT_FROMTYPE : null);